browser/src/Editor/NeovimEditor.tsx

```diff
@@ -465,40 +477,59 @@ export class NeovimEditor extends Editor implements IEditor {
         }
     }
 
-    private _onVimEvent(eventName: string, evt: EventContext): void {
-        UI.Actions.setWindowCursor(evt.windowNumber, evt.line - 1, evt.column - 1)
+    private _updateWindow(currentBuffer: EventContext) {
+        UI.Actions.setWindowCursor(
+            currentBuffer.windowNumber,
+            currentBuffer.line - 1,
+            currentBuffer.column - 1,
+        )
 
         // Convert to 0-based positions
-        this._syntaxHighlighter.notifyViewportChanged(evt.bufferNumber.toString(), evt.windowTopLine - 1, evt.windowBottomLine - 1)
+        this._syntaxHighlighter.notifyViewportChanged(
+            currentBuffer.bufferNumber.toString(),
+            currentBuffer.windowTopLine - 1,
+            currentBuffer.windowBottomLine - 1,
+        )
+    }
 
+    private async _onBufEnter(evt: BufferEventContext): Promise<void> {
+        this._updateWindow(evt.current)
         const lastBuffer = this.activeBuffer
-        const buf = this._bufferManager.updateBufferFromEvent(evt)
```

Comment:
I believe the removal of this line is what is potentially causing the problem. Note that before this line of code was called for _all vim events_ - not just `BufEnter`. It looks like as part of the refactoring we split this out into the BufEnter/BufLeave event handlers, but unfortunately that's not functionality equivalent.

I think for now we'll have to keep the `_onVimEvent` handler for now, but just for the code above:
- the `setWindowCursor` action
- the `syntaxHighlighter.notifyViewportChanged`
- the `bufferManager.updateBufferFromEvent`

Reply:
Ah in which case the line variable is needed before the reducer, ill add that in to the events I think you were likely still right about separating into handlers though as I imagine that as various use cases become know even more handlers will be needed. Im thinking of adding the line to the event handler above the switch statement so it runs before all events